Add extension catalog page and initial contents

nwmac opened the pull request:
### Summary
Fixes #13512

The goal here is to have a new page as part of the extensions documentation - https://extensions.rancher.io/ - where we can list extensions and provide a mechanism for the community to share details of their extensions.

### Occurred changes and/or fixed issues

This PR adds an extensions catalog page to our extension docs.

Each extension is declared in a yaml file and a custom docusaurus plugin generates an index from these.

A custom mdx page renders the catalog page.

This also includes information on how users can submit their extensions to the catalog.

This PR also populates the initial catalog based on Prime and Partner extensions.

### Screenshot/Video

Example of the generated page: 

[ExtensionsCatalog.pdf](https://github.com/user-attachments/files/19250652/ExtensionsCatalog.pdf)

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone <!-- The milestone should automatically be assigned if the linked issue has one, but does not always happen (incorrectly linked, issue has no milestone, etc) -->
- [x] The PR template has been filled out
- [x] The PR has been self reviewed <!-- There are no TODOs, no incorrect files in the PR, all the required files are there, no commented out code, etc-->
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

aalves08 commented on docusaurus/docs/extensions/catalog.mdx:
```diff
@@ -0,0 +1,43 @@
+
+import Catalog from '../../plugins/catalog/Catalog.tsx'
+
+For information on adding your extension to this page, see here. For information of developing your own extension, check out the [developer documentation](http://localhost:3000/extensions/next/extensions-getting-started).
```
should the `here` in `see here` a link? If not, then the whole sentence feels a bit odd since we are telling people to see this page to add extensions to the page... Doesn't it feel a bit odd/weird? 🤔  can't explain it, but something sounds off to me 😛 

Feel free to ignore if you disagree 🙏

aalves08 commented on docusaurus/plugins/catalog/CatalogCard.tsx:
```diff
@@ -0,0 +1,23 @@
+// Catalog card for a single extension
+
+const CatalogCard = (props) => {
+  console.log(props);
```
to clean up

aalves08 reviewed (commented):
Overall I really love the idea and the mechanism is really clever but I wasn't expecting to have the official and partners extensions featured here... I know that they rarely change, but it's yet another area that has to be manually updated every time they "change". 

We're creating another dependency here, that's my only concern.  What do you think?

Added a couple of extra comments

nwmac commented:
@aalves08 Addressed your comments.

For Prime and Partner charts - we can automate this, although to get the link to the maintainer, we would need to request partner authors to ensure they fill out the `maintainers` section of the Helm `Chart.yaml` - see: https://helm.sh/docs/topics/charts/#the-chartyaml-file and also to ensure the `home` property is also set.

I can look at a script to do that.

aalves08 reviewed (approved):
LGTM

aalves08 commented:
> @aalves08 Addressed your comments.
> 
> For Prime and Partner charts - we can automate this, although to get the link to the maintainer, we would need to request partner authors to ensure they fill out the `maintainers` section of the Helm `Chart.yaml` - see: https://helm.sh/docs/topics/charts/#the-chartyaml-file and also to ensure the `home` property is also set.
> 
> I can look at a script to do that.

No worries Neil. We can always iterate over this 🙏  Thank you for getting this in

aalves08 reviewed (approved):
LGTM
